filters/token-amount/test.json

```diff
@@ -0,0 +1,89 @@
+{
+    "type": "token-amount",
+    "test_cases": [
+        {
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": ["in", "at"],
+				"thresholds": [2, 3],
+				"operations": [">=", "<"]
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": true
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": ["in", "played"],
+				"thresholds": [2, 1],
+				"operations": [">=", ">"]
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": false
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": ["in", "played"],
+				"thresholds": [2, 1],
+				"operations": ["==", "=="]
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": true
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": ["in", "played"],
+				"thresholds": 2,
+				"operations": ["==", "<"]
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": true
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": ["in", "played"],
+				"thresholds": 2,
+				"operations": ">="
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": false
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": "in",
+				"thresholds": [1, 3],
+				"operations": [">", "<"]
+            },
+            "inputs": {
+                "sentence": "Andrew played cricket in a soccer stadium in India at 9pm"
+            },
+            "outputs": true
+        },
+		{
+            "class": "TokenAmountFilter",
+            "args": {
+                "keywords": "in",
```

Comment:
Maybe you can do a sanity check for only one keyword with multiple conditions (to check whether there's any contradiction). For example:
thresholds: [3, 1]
operations: [">", "<"]
It would be good to warn the user that the input thresholds/operations would cause a conflict than simply returning a false.

Reply:
I gave it a shot implementing this, not sure if there is a more clean way than just manually constructing the bounds for each keyword but I've tested it with your example as well as:

thresholds: [3, 1]
operations: ["==", "=="]

and raise appropriate ValueErrors for each. Seems like it's working as intended. Let me know in case you have any improvements for the new part of the code.